tiledb/sm/rest/curl.cc

```diff
@@ -180,22 +180,66 @@ static int buffer_list_seek_callback(
   return CURL_SEEKFUNC_FAIL;
 }
 
+/**
+ * This callback function gets called by libcurl as soon as a header has been
+ * received. libcurl buffers headers and delivers only "full" headers, one by
+ * one, to this callback. This callback should return the number of bytes
+ * actually taken care of if that number differs from the number passed to
+ * your callback function, it signals an error condition to the library
+ *
+ * @param res_data points to the delivered data
+ * @param tSize the size of that data is size multiplied with nmemb
+ * @param tCount number of bytes actually taken care of
+ * @param userdata the userdata argument is set with CURLOPT_HEADERDATA
+ * @return
+ */
+
+size_t OnReceiveData(
+    void* res_data, size_t size, size_t count, void* userdata) {
+  size_t length = size * count;
+  size_t index = 0;
+  auto* buffer = static_cast<unsigned char*>(res_data);
+
+  while (index < length) {
```

Comment:
why is this needed? the callback says that it only accepts full headers. we should know that `res_data` is a full header, valid through `length`.

Reply:
I had put it there so I can make sure that the final string does not include `/r` or/and `/n` that are included in the response.